cpp/include/cudf/join.hpp

```diff
@@ -327,5 +327,26 @@ std::unique_ptr<cudf::table> left_anti_join(
   std::vector<cudf::size_type> const& return_columns,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief  Performs a cross join on two tables (left, right)
+ *
+ * The cross join returns the cartesian product of rows from each table.
+ *
+ * The approach is to repeat the left table by the number of rows in the right table
+ * and tile the right table by the number of rows in the left table.
+ *
+ * @throws cudf::logic_error if number of columns in either `left` or `right` table is 0
+ *
+ * @param[in] left                  The left table
+ * @param[in] right                 The right table
+ * @param[in] mr                    Device memory resource to use for device memory allocation
+ *
+ * @returns                         Result of cross joining `left` and `right` tables
```

Comment:
Adding the return type is wrong doxygen. We recently found this out.

Reply:
Another one; is `[in]` really required after we moved to C++ style API? (all of the arguments are `[in]`, we rarely have `[out]`, or `[in,out]` in arguments.) 
IMHO, unless there is a `[out]` in the arguments, it is not necessary to include `[in]` in documentation.

Reply:
It seems like I should leave this `@returns` as-is (I removed the `[in]` bits).

Reply:
Although I agree with relaxing the requirement of writing `[in]`/`[out]`/`[in,out]` in case of APIs that don't need it, I don't know what type of param `mr` should be considered as. @jrhemstad ?